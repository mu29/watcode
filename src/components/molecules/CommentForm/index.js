import React from 'react'
import PropTypes from 'prop-types'
import { Field } from 'redux-form'
import styled from 'styled-components'
import { Button, ReduxField } from 'components'

const Form = styled.form`
  display: flex;
`

const CommentForm = ({
  handleSubmit, isLoading,
}) => (
  <Form onSubmit={ handleSubmit }>
    <Field type="textarea" name="content" placeholder="이 작품은 어땠나요?" component={ ReduxField } />
    <Button type="submit" disabled={ isLoading } width={ 96 } height={ 74 }>작성</Button>
  </Form>
)

CommentForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
}

export default CommentForm
