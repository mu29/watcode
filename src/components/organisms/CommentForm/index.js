import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Field } from 'redux-form'
import { LoadingButton, ReduxField } from 'components'

const Form = styled.form`
  display: flex;
`

const StyledField = styled(Field)`
  padding: 0.5rem;
  font-size: 0.75rem;
  border-radius: 0.125rem 0 0 0.125rem;
`

const StyledButton = styled(LoadingButton)`
  width: 4.5rem;
  height: 3.375rem;
  padding: 0;
  font-size: 0.75rem;
  border-radius: 0 0.125rem 0.125rem 0;
`

const CommentForm = ({
  handleSubmit,
  isLoading,
}) => (
  <Form onSubmit={ handleSubmit }>
    <StyledField type="textarea" name="content" placeholder="이 작품은 어땠나요?" component={ ReduxField } />
    <StyledButton type="submit" isLoading={ isLoading } disabled={ isLoading }>등록</StyledButton>
  </Form>
)

CommentForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
}

export default CommentForm
