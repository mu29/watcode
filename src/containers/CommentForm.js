import React from 'react'
import { compose } from 'recompose'
import { connect } from 'react-redux'
import { reduxForm } from 'redux-form'
import { CommentForm } from 'components'
import { createCommentActions } from 'store/actions'
import { getIsLoading } from 'store/selectors'

const CommentFormContainer = props => <CommentForm { ...props } />

const onSubmit = (data, dispatch, { id, author, email }) => {
  if (!data.content || data.content.length === 0) {
    return
  }

  dispatch(createCommentActions.request({
    id, author, email, ...data,
  }))
}

const mapStateToProps = state => ({
  email: state.auth.user.email,
  author: state.auth.user.displayName,
  isLoading: getIsLoading(state, createCommentActions.type),
})

export default compose(
  connect(mapStateToProps),
  reduxForm({
    form: 'comment',
    onSubmit,
  }),
)(CommentFormContainer)
