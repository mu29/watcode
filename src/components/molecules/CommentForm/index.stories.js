import React from 'react'
import { storiesOf } from '@storybook/react'
import { CommentForm } from 'containers'

storiesOf('Molecules/CommentForm', module)
  .add('기본', () => (
    <CommentForm />
  ))
