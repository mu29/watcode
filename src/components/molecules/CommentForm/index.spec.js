import React from 'react'
import CommentForm from '.'

const attributes = {
  isLoading: false,
  handleSubmit: jest.fn(),
}

const wrap = (props = attributes) => shallow(<CommentForm { ...props } />)

describe('<CommentForm />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = wrap()
    expect(wrapper).toMatchSnapshot()
  })

  it('확인 버튼을 누르면 handleSubmit 함수를 호출한다', () => {
    attributes.handleSubmit.mockClear()
    const wrapper = wrap()
    expect(attributes.handleSubmit).not.toBeCalled()
    wrapper.simulate('submit')
    expect(attributes.handleSubmit).toBeCalled()
  })
})
