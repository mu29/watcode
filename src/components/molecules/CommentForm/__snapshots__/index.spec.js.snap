// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentForm /> 정상적으로 렌더링된다 1`] = `
<Form
  onSubmit={[MockFunction]}
>
  <Field
    component={[Function]}
    name="content"
    placeholder="이 작품은 어땠나요?"
    type="textarea"
  />
  <Button
    disabled={false}
    height={74}
    type="submit"
    width={96}
  >
    작성
  </Button>
</Form>
`;
